File: sorting/selection.java
import java.util.Arrays;

public class selection {
    public static void selectionsort(Comparable[] arr) {
        int N = arr.length;
        for(int i = 0; i<N ; i++ ) {
            int min = i;
            for(int j = i+1 ; j < N ; j++ )
                if (less(arr[j], arr[min])) min = j;
            exch(arr, i, min);
        }
        System.out.println(Arrays.toString(arr));
    }

    private static boolean less(Comparable v, Comparable w) {
        return v.compareTo(w) < 0;
    }

    private static void exch(Comparable[] a, int i, int j) {
        Comparable swap = a[i];
        a[i] = a[j];
        a[j] = swap;
    }

    public static void main(String[] args) {
        selectionsort(new Integer[]{10,9,8,7,6,5,4,3,2,1});
    }
}

File: sorting/selection.c
//gcc selection.c -c; gcc selection.o -o selection; ./selection
#include <stdio.h>
#define MAX_SIZE 10

void selectionsort(int list[], int n) {
    int i, j, temp;

    for(i=0;i<n;i++) {
        int min = i;
        for(j=i+1;j<n;j++){
            if(list[min] > list[j]) {
                min = j;
            }
        }
        temp = list[i];
        list[i] = list[min];
        list[min] = temp;
    }

}

int main() {
    int n = MAX_SIZE;
    int li[n];
    
    for(int i=0; i<n; i++) li[i]=n-i;

    selectionsort(li, n);

    for(int i=0; i<n; i++) {
        printf("%d", li[i]);
    }
    return 0;
}